### PR TITLE
doc improvements

### PR DESCRIPTION
**Pull request by hoebbelsB**
- improved jsdocs:
 - keyCompareMap
 - compareFn
 - distinctUntilSomeChanged
 - selectSlice
- generated new md files
- added new snippet (selecting-the-viewmodel)

### CODE REVIEW

File: libs/state/docs/api/operators/interfaces.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# CompareFn
+
+The function which is used by `distinctUntilSomeChanged` to determine if changes are distinct or not.
```
Comment (BioPhoton) [inline libs/state/docs/api/operators/interfaces.md:3]: also `selectSlices` should get mentioned

File: libs/state/docs/api/operators/select-slice.md [hunk 2]
```diff
@@ -1,7 +1,66 @@
...
+
+```TypeScript
+function selectSlice<T extends object, K extends keyof T, R>(keyCompareMap: KeyCompareMap<T>): OperatorFunction<T, R>
+```
+
+## Parameters
+
+### keyCompareMap
+
+##### typeof: <a href='/docs/generated/operators/interfaces#keycomparemap'>KeyCompareMap</a>&#60;T&#62;
```
Comment (BioPhoton) [inline libs/state/docs/api/operators/select-slice.md:54]: broken link

File: libs/state/docs/snippets/selecting-the-viewmodel.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
+# Selecting the ViewModel
+
+Here are some useful strategies to properly handle `ViewModels` with `@rx-angular`.
```
Comment (BioPhoton) [inline libs/state/docs/snippets/selecting-the-viewmodel.md:3]:
```suggestion
Here are some useful strategies to properly handle `ViewModels` with `@rx-angular/state`.
```

File: libs/state/docs/api/operators/interfaces.md (hunk 1 above)
Comment (hoebbelsB) [reply libs/state/docs/api/operators/interfaces.md]: done

**Pull request merged**